### ISSUE TITLE
Websocket notification when finished uploading is not consistent

### ISSUE THREAD

**Issue by pieroit** [cheshire-cat-ai/core#337]
**Describe the bug**
Sometimes the websocket notification for finished upload is not sent to the admin

**To Reproduce**
Upload a few files via rabbithole from the admin and wait for the notification

**Expected behavior**
Notification for finished upload always comes up

**Comment by Abhijnan-Bajpai** [comment]
Hey! I would like to try working on this. Could I be assigned?

**Comment by pieroit** [comment]
@Abhijnan-Bajpai welcome, tell me if you need help!

Websockets are managed here:
https://github.com/cheshire-cat-ai/core/blob/main/core/cat/routes/websocket.py#L47

This is were the notification is added:
https://github.com/cheshire-cat-ai/core/blob/main/core/cat/rabbit_hole.py#L280

**Comment by varrialeciro** [comment]
After my PR maybe for admin side, besides notification messages in the corner, would be nice if among different status of ingestion, a "spin icon" would appear or a different way to show the progress.

**Comment by pieroit** [comment]
Thanks I'm reading the PR.

Since most of the time is taken by summarization (which is going to be moved to a plugin) and embedding, we can have a percentage of completion of the latest - like embedded 23 out of 52 chunks

**Comment by varrialeciro** [comment]
Yes for summarization should work what you say, but also the splitting phase takes a lot of time, and in this phase there are different file loaders, so could be hard to find a way to count a percentage.

**Comment by pieroit** [comment]
Fixed in #398
